src/tdastro/sources/salt2_jax.py

```diff
@@ -0,0 +1,120 @@
+
+    flux(time, wave) = x0 * [M0(time, wave) + x1 * M1(time, wave)] * exp(c * CL(wave))
+
+    where x0, x1, and c are given parameters, M0 is the average spectral sequence,
+    M1 is the first compoment to describe variability, and CL is the average color
+    correction law.
+
+    We use the formulation in sncosmo where CL is defined such that:
+
+    flux(time, wave) = x0 * [M0(time, wave) + x1 * M1(time, wave)] * 10 ** (-0.4 * c * CL(wave))
+
+    This class is based on the sncosmo implementation at:
+    https://github.com/sncosmo/sncosmo/blob/v2.10.1/sncosmo/models.py
+    The wrapped sncosmo version in sncosmo_models.py is faster and should be used
+    when auto-differentiation is not needed.
+
+    Attributes
+    ----------
+    _m0_model : BicubicInterpolator
+        The interpolator for the m0 parameter.
+    _m1_model : BicubicInterpolator
+        The interpolator for the m1 parameter.
+    _colorlaw : SALT2ColorLaw
+        The data to apply the color law.
+
+    Parameters
+    ----------
+    x0 : parameter
+        The SALT2 x0 parameter.
+    x1 : parameter
+        The SALT2 x1 parameter.
+    c : parameter
+        The SALT2 c parameter.
+    model_dir : `str`
+        The path for the model file directory.
+        Default: ""
+    m0_filename : `str`
+        The file name for the m0 model component.
+        Default: "salt2_template_0.dat"
+    m1_filename : `str`
+        The file name for the m1 model component.
+        Default: "salt2_template_1.dat"
+    cl_filename : `str`
+        The file name of the color law correction coefficients.
+        Default: "salt2_color_correction.dat",
+    **kwargs : `dict`, optional
+        Any additional keyword arguments.
+    """
+
+    def __init__(
+        self,
+        x0=None,
+        x1=None,
+        c=None,
+        model_dir="",
+        m0_filename="salt2_template_0.dat",
+        m1_filename="salt2_template_1.dat",
+        cl_filename="salt2_color_correction.dat",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        # Add the model specific parameters.
+        self.add_parameter("x0", x0, **kwargs)
+        self.add_parameter("x1", x1, **kwargs)
+        self.add_parameter("c", c, **kwargs)
+
+        # Load the data files.
+        self._m0_model = BicubicInterpolator.from_grid_file(
+            os.path.join(model_dir, m0_filename),
+            scale_factor=1e-12,
+        )
+        self._m1_model = BicubicInterpolator.from_grid_file(
+            os.path.join(model_dir, m1_filename),
+            scale_factor=1e-12,
+        )
+
+        # Use the default color correction values.
+        self._colorlaw = SALT2ColorLaw.from_file(os.path.join(model_dir, cl_filename))
+
+    def _evaluate(self, phase, wavelengths, graph_state, **kwargs):
+        """Draw effect-free observations for this object.
+
+        Parameters
+        ----------
+        phase : `numpy.ndarray`
+            A length T array of rest frame timestamps.
```

Comment:
This is the rest frame phases (which is timestamps - time_at_peak). 
We need to add the t0 parameter like in https://github.com/lincc-frameworks/tdastro/blob/main/src/tdastro/sources/sncomso_models.py#L32
so `_evaluate()` is consistent with other physical models as `_evaluate(self, times, wavelengths ...)`